IntegrationTests/TestSuites/Sources/PrimaryTests/main.swift

```diff
@@ -488,21 +490,29 @@ try test("Date") {
 }
 
 // make the timers global to prevent early deallocation
-var timeout: JSTimer?
+var timeouts: [JSTimer] = []
 var interval: JSTimer?
 
 try test("Timer") {
     let start = JSDate().valueOf()
     let timeoutMilliseconds = 5.0
-
+    var timeout: JSTimer!
     timeout = JSTimer(millisecondsDelay: timeoutMilliseconds, isRepeating: false) {
         // verify that at least `timeoutMilliseconds` passed since the `timeout` timer started
         try! expectEqual(start + timeoutMilliseconds <= JSDate().valueOf(), true)
     }
+    timeouts += [timeout]
+
+    timeout = JSTimer(millisecondsDelay: timeoutMilliseconds, isRepeating: false) {
+        fatalError("timer should be cancelled")
+    }
+    timeout = nil
```

Comment:
Totally happy to punt this to another issue, but since there isn’t necessarily a need to cancel  a `setTimeout`, could there be an API that doesn’t require holding the `JSTimer` instance? I guess users could just do 

```swift
JSObject.global.setTimeout(JSOneshotClosure {
  // ...
  return undefined
}, 1000)
```

Reply:
Yes, we can provide such API in this signature for example.

```swift
JSTimer.timeout(millisecondsDelay: 100) {
    // ...
}
```

This PR is already too large, so let's implement it in another PR 👍